pkg/client/client_v2.go

```diff
@@ -22,35 +22,56 @@ import (
 	"github.com/elastic/elastic-agent-client/v7/pkg/utils"
 )
 
-// UnitChangedType defines types for when units are adjusted.
-type UnitChangedType int
+//go:generate stringer -type=ChangeType,Trigger -linecomment -output client_v2_strings.go
+type (
+	// ChangeType defines types for when units are adjusted.
+	ChangeType int
+	// Trigger indicates what triggered a change
+	Trigger int
+)
 
 const (
 	// UnitChangedAdded is when a new unit is added.
-	UnitChangedAdded UnitChangedType = 1
+	UnitChangedAdded ChangeType = iota // unit_added
 	// UnitChangedModified is when an existing unit is modified.
-	UnitChangedModified UnitChangedType = 2
+	UnitChangedModified // unit_modified
 	// UnitChangedRemoved is when an existing unit is removed.
-	UnitChangedRemoved UnitChangedType = 3
+	UnitChangedRemoved // unit_removed
 )
 
-// String returns string representation for the unit changed type.
-func (t UnitChangedType) String() string {
-	switch t {
-	case UnitChangedAdded:
-		return "added"
-	case UnitChangedModified:
-		return "modified"
-	case UnitChangedRemoved:
-		return "removed"
-	}
-	return "unknown"
+const (
+	// TriggerConfig indicates a change in config triggered the change.
+	TriggerConfig Trigger = iota // config_change_triggered
+	// TriggerFeature indicates a change in the features triggered the change.
+	TriggerFeature // feature_change_triggered
+	// TriggerLogLevel indicates a change the log level triggered the change.
+	TriggerLogLevel // log_level_triggered
+	// TriggerStateChange indicates when a unit state has ganged.
+	TriggerStateChange // state_change_triggered
+)
+
+// FeatureFQDN indicates if the FQDN should be used instead of hostname for host.name.
+// See proto.FQDNFeature for details.
+type FeatureFQDN struct {
+	Enabled bool
+}
+
+// Features are configurations for all the features (flag) available.
+// See proto.Features for details.
+type Features struct {
+	FQDN FeatureFQDN
```

Comment:
Lets not added any defined structs for features in the client. That should all come from the protocol, don't want to have to modify the client code to add a new feature.